packages/syft-libs/pyscaffoldext-syft-support/src/pyscaffoldext/syft_support/script.py

```diff
@@ -342,6 +327,33 @@ def main() -> None:
     print(f"\tNot added:{methods_error_count}")
     print("-----------------")
 
+    if write_to_file:
+        DEBUG_FILE_NAME = f"{package_name}.debug.log"
+        PKG_SUPPORT_NAME = f"{package_name}.pkg_support.json"
+
+        with open(PKG_SUPPORT_NAME, "w") as outfile:
+            json.dump(package_support, outfile)
+
+        if debug:
+            with open(DEBUG_FILE_NAME, "w") as f:
+                for item in debug_list:
+                    f.write(f"{item}\n")
+        return
+
+    return json.dumps(package_support)
+
 
 if __name__ == "__main__":
-    main()
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-l", dest="lib", required=True, help="name of the model to be added to ast"
+    )
+    parser.add_argument(
+        "-d", dest="debug", type=int, help="Set it to one to get debug files", default=0
+)
+    args = parser.parse_args()
+
+    DEBUG = args.debug
+    package_name = args.lib
+    generate_package_support(package_name, debug=DEBUG, write_to_file=True)
```

Comment:
Q: Any specific reason of changing the place of the code?

Reply:
creating a argparser at module lvl causes problems with the extension my theory is it is caused as it overwrites / causes problems with the extension's arg parser